myth

```diff
@@ -400,67 +357,39 @@ elif (args.graph) or (args.fire_lasers):
             exitWithError(args.outform, "Error saving graph: " + str(e))
 
     else:
-
-        _issues = []
-        report = Report(args.verbose_report)
-
+        all_issues = []
         for contract in contracts:
-
-            # try:
-            if (args.dynld):
+            if args.dynld:
                 sym = SymExecWrapper(contract, address, dynloader=DynLoader(eth), max_depth=args.max_depth)
             else:
                 sym = SymExecWrapper(contract, address, max_depth=args.max_depth)
-            # except Exception as e:
-            #    exitWithError(args.outform, "Symbolic exection error: " + str(e))
 
             if args.modules:
                 issues = fire_lasers(sym, args.modules.split(","))
             else:
                 issues = fire_lasers(sym)
 
-            if len(issues):
-
-                # Append source code info for SolidityContract
-
-                if (type(contract) == SolidityContract):
-
-                    for issue in issues:
-
-                        if (issue.pc):
-
-                            codeinfo = contract.get_source_info(issue.pc)
-
-                            issue.filename = codeinfo.filename
-                            issue.code = codeinfo.code
-                            issue.lineno = codeinfo.lineno
-
+            if type(contract) == SolidityContract:
                 for issue in issues:
-                    report.append_issue(issue)  # For text and markdown output
-                    _issues.append(issue.as_dict())  # List of dicts for JSON output
-
-        # Finally, output the results
-
-        if (len(_issues)):
-
-            if (args.outform == 'json'):
+                    if issue.pc:
+                        codeinfo = contract.get_source_info(issue.pc)
+                        issue.filename = codeinfo.filename
+                        issue.code = codeinfo.code
+                        issue.lineno = codeinfo.lineno
 
-                result = {'success': True, 'error': None, 'issues': _issues}
-                print(json.dumps(result))
+            all_issues += issues
 
-            else:
-
-                if (args.outform == 'text'):
-                    print(report.as_text())
-                elif (args.outform == 'markdown'):
-                    print(report.as_markdown())
-
-        else:
-            if (args.outform == 'text' or args.outform == 'markdown'):
-                print("The analysis was completed successfully. No issues were detected.")
-            else:
-                result = {'success': True, 'error': None, 'issues': []}
-                print(json.dumps(result))
+        # Finally, output the results
+        report = Report(args.verbose_report)
+        for issue in all_issues:
+            report.append_issue(issue)
+
+        outputs = {
+            'json': report.as_json(),
```

Comment:
I put the json report into `report` followed the other ones `markdown` and `text`, is it OK?